### PR TITLE
docs: improve upgrade path guidance

### PR DESCRIPTION
**Pull request by jkirschner-hashicorp**
### Description

The current upgrade instructions have the following weaknesses:
- They imply more upgrades are needed than necessary (too restrictive with possible paths)
- 1.15 isn't accounted for

### Preview

[Revised version jump instructions page
](https://consul-dazu1a7ur-hashicorp.vercel.app/consul/docs/upgrading/instructions)

### CODE REVIEW

File: website/content/docs/upgrading/upgrade-specific.mdx [hunk 1]
```diff
@@ -515,6 +515,11 @@ to Consul 1.11.11 or later to avoid the breaking nature of that change.
 
 ### Licensing Changes <EnterpriseAlert inline />
 
+You can only upgrade to Consul Enterprise 1.10 from a version of
+Consul Enterprise 1.8 >= 1.8.13 or 1.9 >= 1.9.7. Other versions of
```
Comment (im2nguyen) [inline website/content/docs/upgrading/upgrade-specific.mdx:519]: is this more clear? wasn't sure if it was 1.8.13+ or has to be between 1.8 and 1.8.13

```suggestion
Consul Enterprise 1.8.13+ or 1.9.7+. Other versions of
```

File: website/content/docs/upgrading/instructions/index.mdx [hunk 2]
```diff
@@ -10,45 +10,42 @@ description: >-
...
+upwards to your target version.
+
+For example, to upgrade from Consul 1.12 to Consul 1.15:
+1. Upgrade to Consul 1.14 as an intermediate step.
+   To plan, review the upgrade details for
+   [1.13](/consul/docs/upgrading/upgrade-specific#consul-1-13-x) and
+   [1.14](/consul/docs/upgrading/upgrade-specific#consul-1-14-x).
+2. Upgrade to Consul 1.15.
+   To plan, review the upgrade details for
+   [1.15](/consul/docs/upgrading/upgrade-specific#consul-1-15-x).
```
Comment (im2nguyen) [inline website/content/docs/upgrading/instructions/index.mdx:30]:
```suggestion
For example, to upgrade from Consul 1.12 to Consul 1.15:

1. Upgrade to Consul 1.14 as an intermediate step.
   To plan, review the upgrade details for
   [1.13](/consul/docs/upgrading/upgrade-specific#consul-1-13-x) and
   [1.14](/consul/docs/upgrading/upgrade-specific#consul-1-14-x).
1. Upgrade to Consul 1.15.
   To plan, review the upgrade details for
   [1.15](/consul/docs/upgrading/upgrade-specific#consul-1-15-x).
```

File: website/content/docs/upgrading/instructions/index.mdx [hunk 3]
```diff
@@ -10,45 +10,42 @@ description: >-
...
+The following table provides links to dedicated instructions
+for directly upgrading from a version in the starting range
+to a destination version.
+
+| Starting Version Range | Destination Version | Upgrade Instructions |
+| ---------------------- | ------------------- | -------------------- |
+| 1.8.0 - 1.9.17         | 1.10.12             | Follow the [instructions for upgrading to final 1.10.x](/consul/docs/upgrading/instructions/upgrade-to-1-10-x) |
+| 1.6.9 - 1.8.18         | 1.8.19              | Follow the [instructions for upgrading to final 1.8.x](/consul/docs/upgrading/instructions/upgrade-to-1-8-x) |
+| 1.2.4 - 1.6.9          | 1.6.10              | Follow the [instructions for upgrading to final 1.6.x](/consul/docs/upgrading/instructions/upgrade-to-1-6-x) |
+| 0.8.5 - 1.2.3          | 1.2.4               | Follow the [instructions for upgrading to final 1.2.x](/consul/docs/upgrading/instructions/upgrade-to-1-2-x) |
```
Comment (im2nguyen) [inline website/content/docs/upgrading/instructions/index.mdx:43]: super small nit -- we should keep the link as close to the actual page title as possible. also reducing the text so everything fits on one line

```suggestion
| 1.8.0 - 1.9.17         | 1.10.12             | Refer to [Upgrading to latest 1.10.x](/consul/docs/upgrading/instructions/upgrade-to-1-10-x) |
| 1.6.9 - 1.8.18         | 1.8.19              | Refer to [Upgrading to latest 1.8.x](/consul/docs/upgrading/instructions/upgrade-to-1-8-x) |
| 1.2.4 - 1.6.9          | 1.6.10              | Refer to [Upgrading to latest 1.6.x](/consul/docs/upgrading/instructions/upgrade-to-1-6-x) |
| 0.8.5 - 1.2.3          | 1.2.4               | Refer to [Upgrading to latest 1.2.x](/consul/docs/upgrading/instructions/upgrade-to-1-2-x) |
```

image

**Review by im2nguyen — APPROVED** [review]

**Review by eculver — APPROVED** [review]
LGTM.

File: website/content/docs/upgrading/upgrade-specific.mdx (hunk 1 above)
Comment (jkirschner-hashicorp) [reply website/content/docs/upgrading/upgrade-specific.mdx:519]: I agree that the current wording isn't clear (copied it from another page).

I'm concerned that the proposed change might imply that 1.9.5 is acceptable, when it really isn't. 1.9.5 is technically "1.8.13+" ... but it won't work. If you're on the 1.9 series, you need 1.9.7+. If you're on the 1.8 series, you need 1.8.13+.

Comment (jkirschner-hashicorp) [reply website/content/docs/upgrading/upgrade-specific.mdx:519]: Changing to:

You can only upgrade to Consul Enterprise 1.10 from the following Enterprise versions:
- 1.8 release series: 1.8.13+
- 1.9 release series: 1.9.7+

**Pull request merged**